File: scaffolds/nodejs/src/services/health.service.ts
import { env } from '../config/env.js';

export interface HealthStatus {
  status: 'healthy' | 'degraded' | 'unhealthy';
  version: string;
  uptime: number;
  timestamp: string;
  // Extend with dependency checks as your service grows:
  // dependencies?: Record<string, 'ok' | 'error'>;
}

export const getHealthStatus = async (): Promise<HealthStatus> => ({
  status: 'healthy',
  version: env.SERVICE_VERSION,
  uptime: Math.floor(process.uptime()),
  timestamp: new Date().toISOString(),
});

File: scaffolds/fastapi/app/services/health_service.py
import time
from datetime import datetime, timezone

from app.core.config import get_settings
from app.models.schemas.health import HealthData

_start_time = time.time()


async def get_health_status() -> HealthData:
    settings = get_settings()
    return HealthData(
        status="healthy",
        version=settings.SERVICE_VERSION,
        uptime=round(time.time() - _start_time, 2),
        timestamp=datetime.now(timezone.utc).isoformat(),
        # Add dependency checks as your service grows:
        # dependencies={"database": await check_db(), "cache": await check_cache()},
    )
